R/hn_user.R

```diff
@@ -0,0 +1,56 @@
+
+#' @title Print for "hn_user" type objects
+#'
+#' @param x "hn_user" type list
+#'
+#' @param ... further arguments passed to or from other methods
+#'
+#' @export
+#'
+#' @examples
+#'
+#' hacker_user <- get_user_by_username("steveddaniels")
+#' print(hacker_user)
+
+print.hn_user <- function(x, ...) {
+  utils::str(x)
+}
+
+
+#' Creates an object representing Hacker News user
+#'
+#' @param user list returned by Hacker News API
+#' via get_user_by_username function
+#'
+#' @return object represanting a hn_user
+#'
+create_hn_user <- function(user) {
+  if (is.null(user)) {
```

Comment:
Could you explain me what is this check used for?

Reply:
to avoid passing NULL to structure. If you pass NULL, R shows a warning.

Reply:
Yes, but we expect a proper R object to be created that passes the `hn_user` validation. Is there an instance when NULL is passed to this funcion?

Reply:
get_user_by_username('xyz')
.send_request returns NULL and passes it to create_hn_user

Reply:
Good catch! I did not expect the API to return a 200 code when a user does not exist. We should implement a validation system that will check if the response is not empty.